src/Repository/JobsRepository.php

```diff
@@ -20,47 +20,125 @@ public function __construct(ManagerRegistry $registry)
     {
         parent::__construct($registry, Jobs::class);
     }
-
-    public function save(Jobs $entity, bool $flush = false): void
+    
+    /**
+     * save
+     *
+     * @param  mixed $entity
+     * @return bool
+     */
+    public function save(Jobs $entity): bool
     {
         $this->getEntityManager()->persist($entity);
+        $this->getEntityManager()->flush();
 
-        if ($flush) {
-            $this->getEntityManager()->flush();
-        }
+        //if previous flush method is succesfull the $entity->getId() will return the id of the inserted company
+        //if previous flush method is not succesfull $entity->getId() will return NULL
+        return $entity->getId() > 0; // if null willreturn false
     }
-
-    public function remove(Jobs $entity, bool $flush = false): void
+    
+    /**
+     * remove
+     *
+     * @param  mixed $entity
+     * @return void
+     */
+    public function remove(Jobs $entity): void
     {
         $this->getEntityManager()->remove($entity);
+        $this->getEntityManager()->flush();
+    }
+       
+    /**
+     * update
+     *
+     * @param  mixed $id
+     * @param  mixed $params
+     * @return int
+     */
+    public function update(int $id, array $params): int
+    {
+        //create a new query builder object
+        $queryBuilder = $this->createQueryBuilder('j');
+
+        //prepare an update statement
+        $nbUpdatedRows = $queryBuilder->update()
+            ->set('j.name', ':jobName')
+            ->set('j.description', ':jobDescription')
+            ->set('j.active', ':jobActive')
+            ->set('j.priority', ':jobPriority')
+            ->where('j.id = :jobId')
+            ->setParameter('jobName', $params['name'])
+            ->setParameter('jobDescription', $params['description'])
+            ->setParameter('jobActive', $params['active'])
+            ->setParameter('jobPriority', $params['priority'])
+            ->setParameter('jobId', $id)
+            ->getQuery()
+            ->execute();
 
-        if ($flush) {
-            $this->getEntityManager()->flush();
-        }
+        return $nbUpdatedRows;
+    }
+    
+    /**
+     * getJobName
+     *
+     * @param  mixed $name
+     * @return array
+     */
+    public function getJobName(string $name): array
+    {
+        $queryBuilder = $this->createQueryBuilder('j');
+        $company = $queryBuilder
+            ->select('j.id, j.name')
```

Comment:
remove this select only for name because we need all the information about the job and not only one field. If you remove this specific name select the query will return also the join with company result and the available company information.